travis: separate check patch and age of git base

marc-hb opened the pull request:
As requested in fix for checkpatch issue PR #2543 which will be rebased
on top of this.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

plbossart commented on .travis.yml:
```diff
@@ -81,6 +85,17 @@ jobs:
+        #
+        # FIXME: this addresses only pull requests without any merge
+        # submitted as part of the pull request itself.
+        - *unshallow_for_merge_base
```
out of curiosity, what are those '*unshallow_for_merge_base" and "&unshallow_for_merge_base"?

marc-hb replied:
They're YAML anchors = how to avoid copy/paste/diverge.

kv2019i commented:
@marc-hb Shouldn't the checkpatch run https://travis-ci.org/github/thesofproject/linux/builds/740309390 succeed here (i.e. detect git clone is too shallow and run unshallow on the repo before checkpatch)...?

marc-hb commented:
> @marc-hb Shouldn't the checkpatch run succeed here (i.e. detect git clone is too shallow and run unshallow on the repo before checkpatch)...?

Yes it should. But it doesn't. @ranj063 spotted a similar issue in https://travis-ci.org/github/thesofproject/linux/jobs/739977895.

I reproduced this locally at last and root-caused it to the crazy (buggy?) way `git fetch` works in shallow clones. I'm drafting a bug right now to document this.

marc-hb commented:
Rebased and added `--all` option to `git merge-base` command to help (a bit) with shallowness issues mentioned above.

https://travis-ci.org/github/thesofproject/linux/builds/741026106 is looking good.

Even though the #2543 "fix" is uncertain and needs to be discussed and revisited I think this PR is good because it has no real functional change, it's just removing a confusing "optimization" and making the Travis output simpler and clearer.

PS: ICL_RVP_HDA's failure to boot in  https://sof-ci.01.org/linuxpr/PR2547/build4824/devicetest can obviously not be related to this.

marc-hb commented:
> I reproduced this locally at last and root-caused it to the crazy (buggy?) way git fetch works in shallow clones. I'm drafting a bug right now to document this.

Finally done, see graphical explanation of the issue in freshly filed `Shallow cloning in CI can create pull requests with one million commits` #2556

plbossart commented:
Not sure where this is going and if Travis is still relevant. closing.

marc-hb commented:
Summary at https://github.com/thesofproject/linux/pull/2543#issuecomment-765029638
